src/base.jl

```diff
@@ -41,13 +41,15 @@ struct System <: PowerSystemType
     branches::Union{Nothing, Vector{<:Branch}}
     storage::Union{Nothing, Vector{<:Storage}}
     basepower::Float64 # [MVA]
-    time_periods::Int64
+    forecasts::Union{Nothing, SystemForecasts}
+    services::Union{Nothing, Vector{ <: Service}}
+    annex::Union{Nothing,Dict{Any,Any}}
 
     function System(buses, generators, loads, branches, storage_devices, basepower,
-                    time_periods; kwargs...)
+                    forecasts, services, annex; kwargs...)
```

Comment:
I didn't see this earlier. Are we sure they will be no issue when the inner constructor takes a `kwargs...`?

Reply:
Also, in the inner constructor we are not propagating `kwargs...` any further. We can use `validate=false` here instead, but there are tradeoffs. We will have to use `pop!` instead of `get` for using values from `kwargs` when we don't want to it propagate further, and still use `get` when we do. This requires a clearer understanding of all dispatch methods. On the other hand we'll catch errors if someone types in an incorrect keyword. At the moment if the user makes a typo, the functions will silently choose a default value that might be related what they intend to do.

Reply:
@kdheepak is right, inner constructors should take only specific keyword argument. Also, I am not sure that the checks should be optional. We should move some of them to PowerSimulations but not all of them are needed here.